acoustid/future/fpindex/updater.py

```
@@ -0,0 +1,246 @@
+
+
+class ShutdownError(Exception):
+    pass
+
+
+def shutdown(event: asyncio.Event) -> None:
+    logger.info("Shutting down...")
+    event.set()
+
+
+async def replicate_from_pg() -> None:
```

Ruff (0.8.2)

`replicate_from_pg` is too complex (23 > 10)

(C901)